arch/Kconfig

```diff
@@ -114,6 +115,7 @@ config RISCV
 	bool
 	select ARCH_IS_SET
 	select ARCH_SUPPORTS_COREDUMP
+	select ARCH_SUPPORTS_COREDUMP_PRIV_STACKS
```

Comment:
`select ARCH_SUPPORTS_COREDUMP_PRIV_STACKS` needs to also be added to `config XTENSA` right?

Reply:
On Xtensa, coredump is only enabled by the SoC or board. So this is not done via arch kconfig.

Reply:
I have moved the kconfig to the SoC instead of board, which is more appropriate.